src/Microsoft.DotNet.CodeAnalysis/Analyzers/ResourceUsageAnalyzer.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.DotNet.CodeAnalysis.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ResourceUsageAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Invalid SR.Format call";
+        private const string Description = "The SR.Format call should be removed.";
+        private const string AnalyzerName = "ResourceUsageAnalyzer";
+
+        private static DiagnosticDescriptor InvalidSRFormatCall = new DiagnosticDescriptor(DiagnosticIds.BCL1000.ToString(), Title, "", AnalyzerName, DiagnosticSeverity.Error, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get { return ImmutableArray.Create(InvalidSRFormatCall); } }
+
+        INamedTypeSymbol SRSymbol { get; set; }
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(StartCompile);
+        }
+
+
+        private void StartCompile(CompilationStartAnalysisContext obj)
+        {
+            SRSymbol = obj.Compilation.GetTypeByMetadataName("System.SR");
```

Comment:
I was trying to be consistent with Roslyn model... since we are at the semantic model I chose to use symbol here.